pkgs/by-name/sy/syzkaller/package.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  go,
+  ncurses,
+}:
+let
+  rev = "67d7ec0a69193b2acbb8cdb5109d5505a9128ac5";
```

Comment:
There is no practical use in doing this since we can reference the attributes in the derivation with `finalAttrs` (see comments, below).

```suggestion
```

Reply:
Out of curiosity: Is there any practical reason for favoring finalAttrs over a let binding?

Reply:
It's better for overridability AFAIC (see https://github.com/NixOS/nixpkgs/issues/315337#issuecomment-2138814370). This might be particularly bad in this case since if we override `src`, the `postPatch` phase will continue using the old `rev` from the `let in` block and this isn't what we want.

We can actually store `commitDate` and `dateString` inside the derivation itself, using `passthru`:

```nix
passthru = {
  commitDate = "2024-01-09";
  dateString = lib.strings.concatStringsSep "-" [
    (lib.strings.replaceChars [ "-" ] [ "" ] finalAttrs.passthru.commitDate)
    "133700" # dummy time
  ];
};
```
 
Then access them with e.g. `finalAttrs.passthru.dateString`, but I don't know much this pattern is accepted in nixpkgs.

Reply:
Thanks for the detailed explanation! Reading the comment, this clearly makes a lot of sense.

Re the passthru pattern; I'm not a huge fan of this, as it exposes these values that are quite internal to the package build in a way that a user might think it's meant to be consumed by him. I think we have a good compromise with how it currently is.

Reply:
Yes, I very much agree.